packages/compass-query-bar/src/components/query-history-button-popover.tsx

```diff
@@ -64,6 +64,8 @@ const QueryHistoryButtonPopover = ({
     [onOpenPopover]
   );
 
+  const closePopover = useCallback(() => setIsOpen(false), [setIsOpen]);
```

Comment:
Why are we changing the behavior for update selected but not for the other saved queries types?

Reply:
We decided to not add a breaking change, as there is a project in the future for reworking the query bar and this popover will be re-designed. The only reason we are closing here is because, due to some leafy-green hardcoded z-indexes, part of the popup shows over the modal.

Reply:
Hmm, if we're not planning to change the current behavior, then you'd definitely want to separate card click actions and open update modal actions in some way because right now it's implemented in a way where opening a recent query is not closing the popover, but opening a favorite (even without `update`) will close it which is just weirdly inconsistent.

Saying that though, I think chaging this popup to autoclose when item is selected _in all cases_ was on the list of things to do for awhile. @Anemy maybe you remember this and even haave a ticket reference?

Reply:
If this is the case and there is a ticket, it could be handled in the backlog when the ticket is prioritised. Unless there is a big reason, apart from that the behaviour changed and it has been approved, I would keep the behaviour in the PR and continue with the plan of tackling it during the query bar improvements or in that specific ticket.

Reply:
Sure, then we just need to fix the favorited queries behavior without the update modal showing up that this PR currently breaks 👍

Reply:
Ah yes, I was testing your suggestion to use the localAppRegistry instead of the globalAppRegistry for opening the update modal, but it doesn't work because the communication has to be between plugins, so the only solution is to use the globalAppRegistry.

I'll revert that change and hopefully evergreen will be green 😄 .